sql/src/test/java/io/crate/auth/user/UserTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.auth.user;
+
+import io.crate.analyze.user.Privilege;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import java.util.Set;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class UserTest {
+
+    @Test
+    public void test_user_has_al_cluster_privilege() {
+        var grantALOnCluster = new Privilege(
+            Privilege.State.GRANT,
+            Privilege.Type.AL,
+            Privilege.Clazz.CLUSTER,
+            null,
+            "crate"
+        );
+        var arthur = new User("arthur", Set.of(), Set.of(grantALOnCluster), null);
+        assertThat(
+            arthur.hasPrivilege(Privilege.Type.AL, Privilege.Clazz.CLUSTER, null, "hitchhikers"),
+            Matchers.is(true));
```

Comment:
do we really need this test? privilege matching is already tested, adding a type doesn't require in additional tests afaik. otherwise we could maybe merge it into `PrivilegesTest`

Reply:
I wasn't sure if any changes are required so I added it beforehand - can remove it again since there weren't any changes required.